### PR TITLE
Don't throw and exception when raise on error is set to false

### PR DESCRIPTION
**Pull request by Tasty213**
As raised in #276  and in compliance with the [specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md#detecting-resource-information-from-the-environment) make it so that running the detect method on a detector with `raise_on_error` set to false doesn't throw an exception if it can't find any GCP resources.

### CODE REVIEW

**Comment by aabmass** [comment]
@Tasty213 can you sign the CLA?

**Comment by Tasty213** [comment]
I think I've already done so? The check shows as passed

**Comment by Tasty213** [comment]
fixed the linter error

**Comment by Tasty213** [comment]
Could've sworn the longer passed locally

**Comment by aabmass** [comment]
> Could've sworn the linter passed locally

Could be some weird flake in the CI setup. Lmk if you have trouble getting it to pass in CI and I can try to fix it

**Comment by Tasty213** [comment]
Figured out what was wrong with my local testing, I use Windows and have git to checkout CRLF and commit LF, so when I ran the linter locally every single file threw a warning as it was in the 'wrong' line ending style giving pylint [unexpected-line-ending-format](https://pylint.pycqa.org/en/latest/user_guide/messages/convention/unexpected-line-ending-format.html). I've set pycharm to force everything in the project to LF but now git reckons I've changed every file (until I try and stage and it realises I haven't).

Either this should now pass the lint step.

File: opentelemetry-resourcedetector-gcp/src/opentelemetry/resourcedetector/gcp_resource_detector/__init__.py [hunk 1]
```diff
@@ -199,5 +199,8 @@ def detect(self) -> "Resource":
                     self.gcp_resources = found_resources
                     break
         if not self.gcp_resources:
-            raise NoGoogleResourcesFound()
+            if self.raise_on_error:
```
[the remark this answers is not in the document]
Comment (Tasty213) [reply opentelemetry-resourcedetector-gcp/src/opentelemetry/resourcedetector/gcp_resource_detector/__init__.py]: yeah that'd make more sense

**Review by aabmass — APPROVED** [review]
Thank you for contributing!

**Pull request merged**